### PR TITLE
fix: Wait until operator remove cluster scope objects itself

### PR DESCRIPTION
**Pull request by tolusha**
Signed-off-by: Anatolii Bazko <abazko@redhat.com>

### What does this PR do?
Wait until operator remove cluster scope objects itself

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/18501

### How to test this PR?
1. deploy eclipse-che  on OpenShift
2. run `server:delete` make sure when checluster is removed then all cluster scope objects are removed as well

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: src/tasks/installers/operator.ts [hunk 1]
```diff
@@ -361,16 +361,34 @@ export class OperatorTasks {
     {
       title: `Delete the Custom Resource of type ${CHE_CLUSTER_CRD}`,
       task: async (_ctx: any, task: any) => {
+        await kh.deleteCheCluster(flags.chenamespace)
+        do {
+          await cli.wait(20 * 60 * 1000) // default timeout in che operator to clean up cluster scope objects
```
Comment (mmorhun) [inline src/tasks/installers/operator.ts:366]: This line adds 2 minutes of delay in any case. I would avoid such a delay if it possible

Comment (AndrienkoAleksandr) [reply src/tasks/installers/operator.ts:366]: That's not possible... che-operator needs time to clean up created resources any way...

File: src/tasks/installers/operator.ts [hunk 2]
```diff
@@ -361,16 +361,34 @@ export class OperatorTasks {
...
+        } while (await kh.getCheCluster(flags.chenamespace))
```
Comment (AndrienkoAleksandr) [inline src/tasks/installers/operator.ts:367]: Sorry, I'm afraid chectl could hangs here...

File: src/tasks/installers/operator.ts (hunk 1 above)
Comment (tolusha) [reply src/tasks/installers/operator.ts]: It was my mistake. I changed to 20 sec

File: src/tasks/installers/operator.ts (hunk 2 above)
Comment (tolusha) [reply src/tasks/installers/operator.ts]: Tested. With a fake finalizer.
checluster is removed when finalizers are removed by patching CR
```
await kh.patchCheClusterCustomResource(checluster.metadata.name, flags.chenamespace, { metadata: { finalizers: null } })
```

**Review by mmorhun — APPROVED** [review]

**Review by flacatus — APPROVED** [review]

**Review by AndrienkoAleksandr — APPROVED** [review]

**Pull request merged**